mempool/src/pool/tests.rs

```diff
@@ -1879,3 +1879,88 @@ async fn no_empty_bags_in_indices(#[case] seed: Seed) -> anyhow::Result<()> {
     mempool.store.assert_valid();
     Ok(())
 }
+
+struct TestTxAccumulator {
+    txs: Vec<SignedTransaction>,
+    total_size: usize,
+    target_size: usize,
+}
+
+impl TestTxAccumulator {
+    fn new(target_size: usize) -> Self {
+        Self {
+            txs: Vec::new(),
+            total_size: 0,
+            target_size,
+        }
+    }
+}
+
+impl TransactionAccumulator for TestTxAccumulator {
+    fn add_tx(&mut self, tx: SignedTransaction) {
+        self.total_size += tx.encoded_size();
```

Comment:
The solution to that problem I mentioned in the other comment is to add the new size to a temporary variable, then only push if the variable doesn't exceeds the size limit, otherwise switch "done" to true. 

This, though, introduces the limitation that the mempool should be smart about adding transaction by trying to pack more at the end when the available size becomes smaller and smaller, which is an optimization problem we can worry about later. This problem can be irrelevant if the size of transactions is much smaller than the block size limit, which can be an assumption we accept, but have to acknowledge.